src/display/canvas.js

```diff
@@ -1359,9 +1359,7 @@ var CanvasGraphics = (function CanvasGraphicsClosure() {
       }
 
       var name = fontObj.loadedName || 'sans-serif';
-      var bold = fontObj.black ? (fontObj.bold ? '900' : 'bold') :
-                                 (fontObj.bold ? 'bold' : 'normal');
-
+      var bold = fontObj.black ? '900' : (fontObj.bold ? 'bold' : 'normal');
```

Comment:
In practice this assumption seems to give better, i.e. closer to Adobe Reader, rendering in PDF.js for me locally on Windows (even if it doesn't manifest itself on the bots).
It actually seems that Adobe Reader might simply treat (non-embedded) ArialBlack-Bold the same as ArialBlack.